File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());
        int[] nums = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0;i<n;i++){
            nums[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(nums);
        int m = Integer.parseInt(br.readLine());
        int[] find = new int[m];
        st = new StringTokenizer(br.readLine());
        for(int i=0;i<m;i++){
            int target = Integer.parseInt(st.nextToken());
            bw.write(binarySearch(nums,target)?"1\n" : "0\n");
        }
        
        bw.flush();
        bw.close();
        br.close();
    }
    private static boolean binarySearch(int[] arr, int target){
        int left = 0;
        int right = arr.length-1;
        while(left<=right){
            int mid = left+(right-left)/2;
            if(arr[mid]==target){
                return true;
            }else if(arr[mid]<target){
                left = mid+1;
            }else{
                right = mid-1;
            }
        }
        return false;
    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input  = sys.stdin.readline

n = int(input())
arr = list(map(int,input().split()))
arr.sort()
m = int(input())
brr = list(map(int,input().split()))
res =[]

for i in range(m):
    find = False
    target = brr[i]
    start = 0
    end = len(arr)-1
    while start<=end:
        mid = int((start+end)/2)
        midv = arr[mid]
        if midv>target:
            end = mid-1
        elif midv<target:
            start = mid+1
        else:
            find =True
            break
    if find:
        print(1)
    else:
        print(0)